lib/parse.js

```diff
@@ -18,7 +18,9 @@ var defaults = {
 
 var parseValues = function parseQueryStringValues(str, options) {
     var obj = {};
-    var parts = str.split(options.delimiter, options.parameterLimit === Infinity ? undefined : options.parameterLimit);
+    var cleanStr = options.ignoreQueryPrefix && str.slice(0, 1) === '?' ? str.slice(1) : str;
```

Comment:
`str.charAt(0) === '?'` might be cleaner than the slice(0, 1) here - alternatively, `options.ignoreQueryPrefix ? str.replace(/^\?/, '') : str`?

Also, please parenthesize the ternary conditional since there's multiple items.